yarn-project/stdlib/src/logs/directional_app_tagging_secret.ts

```diff
@@ -0,0 +1,61 @@
+import { Grumpkin, poseidon2Hash } from '@aztec/foundation/crypto';
+import { type Fq, Fr, type Point } from '@aztec/foundation/fields';
+
+import type { AztecAddress } from '../aztec-address/index.js';
+import type { CompleteAddress } from '../contract/complete_address.js';
+import { computeAddressSecret, computePreaddress } from '../keys/derivation.js';
+
+export class DirectionalAppTaggingSecret {
+  private constructor(public readonly value: Fr) {}
+
+  /**
+   * Derives shared tagging secret and from that, the app address and recipient derives the directional app tagging
+   * secret.
+   *
+   * @param localAddress - The complete address of entity 1 in the shared tagging secret derivation scheme
+   * @param localIvsk - The incoming viewing secret key of entity 1
```

Comment:
Compared to the original code I prefixed these 2 with "local" to communicate that they correspond to 1 entity. Then I think it's clearer that external corresponds to the entity 2.